### PR TITLE
Bump plutus to 52b23b8

### PR DESCRIPTION
**Pull request by sorki**
And bunch of its dependencies, mostly synchronized with
`cabal.project` in Plutus repo.

* `IsData` -> `To/FromData`
* `Contract`s -> `Promise`s
* PAB BuiltIn
* Adding `traceIfFalse` so we get *two* error messages!
* `OverloadedStrings`

Disabled "Expose 'lock' endpoint, but not 'guess' endpoint" test
since both are now exposed from start.

### CODE REVIEW

**Comment by sorki** [comment]
@luigy tried to fix the endpoints thingie here, but I'm not quite sure about it.

**Comment by luigy** [comment]
@sorki nice! are you seeing the expected behavior with the new `select` changes? from a skim to the new select api PR it doesn't seem like it would change the issue from https://github.com/input-output-hk/plutus/issues/3526 just yet, but I'll have to take a closer look. 
Once this is ready we should put out a new devcontainer release from plutus and point to it

**Comment by sorki** [comment]
Simplified variant does work, reverting my attempt at 0bee7eaeaf3b8aeb8370db1cadc93ddbc533532b for now as that interestingly causes both endpoints to not be available again (but the trace seems happy).

In this case it might be better to simply expose both anyways, as you can now make guess before locking and everything just works.

cc https://github.com/input-output-hk/plutus/issues/3526

File: examples/test/Spec/Game.hs [hunk 1]
```diff
@@ -36,12 +36,15 @@ theContract = game
 -- No funds locked, so W2 (and other wallets) should not have access to guess endpoint
 tests :: TestTree
 tests = testGroup "game"
-    [ checkPredicate "Expose 'lock' endpoint, but not 'guess' endpoint"
-        (endpointAvailable @"lock" theContract t1
-          .&&. not (endpointAvailable @"guess" theContract t1))
-        $ void $ Trace.activateContractWallet w1 (lock @ContractError)
+    [
+-- Note: Used to be the case prior Promises but now we expose both at once
```
Comment (sorki) [inline examples/test/Spec/Game.hs:40]: Left a note here about endpoint behavior. Might be worth expanding it a bit / referencing our discussion.

**Comment by silky** [comment]
Thanks for putting this together @sorki ; sorry it dropped of the radar a bit, but it was helpful for me for preparing v1.0.7 and merging #26 !

**Comment by sorki** [comment]
> Thanks for putting this together @sorki ; sorry it dropped of the radar a bit, but it was helpful to me for preparing v1.0.7 and merging #26 !

You're welcome! I mostly wanted to try updating the thing to learn about the process a little and to test some assumptions, glad it was helpful.

**Pull request closed without merging**